ttnn/cpp/ttnn/operations/ccl/barrier/barrier.hpp

```diff
@@ -0,0 +1,22 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "ttnn/cpp/ttnn/operations/ccl/ccl_host_types.hpp"
+
+#include "ttnn/cpp/ttnn/operations/ccl/barrier/device/barrier_op.hpp"
```

Comment:
This include must be moved to cpp. It really does not belong in this header. 
Presence of such includes makes our build times very long.

Reply:
Agreed, have replaced it with just the decorators library that was needed